CMakeLists.txt

```diff
@@ -1,22 +1,34 @@
-# 3.15 is the minimum.
-# 3.17 for NVC++.
-# 3.18 for C++17 + CUDA.
+# 3.15 is the minimum for including the project with add_subdirectory.
+# 3.21 is the minimum for the developer build.
 cmake_minimum_required(VERSION 3.15)
 
 # Determine whether libcudacxx is the top-level project or included into
 # another project via add_subdirectory().
-if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_LIST_DIR}")
-  set(LIBCUDACXX_TOPLEVEL_PROJECT ON)
+enable_language(CXX) # For CMAKE_CXX_COMPILER_ID
```

Comment:
Calling `enable_language` before `project` is undefined behavior and shouldn't be done.

Reply:
I'm not sure any of the hacking we do to make `lit` generate a config is required. We might be able to just 'be normal' and have our project at the top.

Reply:
It looks like we can just move this into the project call once the `CMAKE_CUDA_HOST_COMPILER` hack is removed 👍